src/coreclr/src/jit/lclmorph.cpp

```diff
@@ -809,6 +809,15 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
             return;
         }
 
+#ifdef _TARGET_X86_
+        if (m_compiler->info.compIsVarArgs && varDsc->lvIsParam && !varDsc->lvIsRegArg)
+        {
+            // TODO-ADDR: For now we ignore all stack parameters of varargs methods,
```

Comment:
`fgMorphStackArgForVarArgs ` is not called at all in this case, is it?
With your change that tree looks like `LCL_FLD_ADDR` instead of `ADDR->FIELD->ADDR->LCL_VAR` and `fgMorphStackArgForVarArgs` is called only for `LCL_VAR` and `LCL_FLD`.
So as I understand the issue is to find the right place to call it, the function itself can handle it (it takes only `lclNum` and `offset`, so it doesn't care about the tree structure), am I correct or missing something?

Reply:
Well, the function does take and offset but it also produces and indir in all cases and that's not going to be correct for local address nodes.

And yes, the problem is finding the right place where to do this transform. I was working on merging this function with `fgMorphImpliciByRefArgs` when I ran into this issue and it turns out that these 2 are unnecessarily different in this regard:
* `fgMorphStackArgForVarArgs` is done in preorder, when encountering `LCL_VAR` and `LCL_FLD` nodes.
* `fgMorphImpliciByRefArgs` is also done in preorder but when encountering nodes that use `LCL_VAR` and `LCL_FLD` nodes, in "pre-pre-order" so to speak

The only difference between these 2 is the way the address is computed, there doesn't seem to be any good reason for any other differences to exist in their morphing. Especially considering that the stack varargs case is always going to suffer from poor coverage.